amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/parser/spec/domain/OasLikeEndpointParser.scala

```diff
@@ -141,11 +141,11 @@ abstract class OasEndpointParser(entry: YMapEntry, producer: String => EndPoint,
         val idCounter = new IdCounter()
         val securedBy = entry.value
           .as[Seq[YNode]]
-          .map(s => OasLikeSecurityRequirementParser(s, endpoint.withSecurity, idCounter).parse())
-          .collect { case Some(s) => s }
+          .flatMap(s =>
+            OasLikeSecurityRequirementParser(s, (se: SecurityRequirement) => se.adopted(endpoint.id), idCounter)
+              .parse())
 
-        if (securedBy.nonEmpty)
-          endpoint.set(OperationModel.Security, AmfArray(securedBy, Annotations(entry.value)), Annotations(entry))
+        endpoint.set(OperationModel.Security, AmfArray(securedBy, Annotations(entry.value)), Annotations(entry))
```

Comment:
Shouldn't we check if it is empty? Why omit that?

Reply:
If security key is present, field should be set, regarding if there is some value or not.
Not have the key present is not the same that have it without values.
New, the secured by is being set always that the key is present, wherever the list is empty or not.